src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -170,12 +166,14 @@ private void PopulateRouteData(HttpContext context)
 
         private async Task<bool> AuthenticateAndAuthorizeAsync(HttpContext context, FunctionDescriptor descriptor)
         {
-            if (!descriptor.Metadata.IsProxy)
-            {
-                var policyEvaluator = context.RequestServices.GetRequiredService<IPolicyEvaluator>();
-                AuthorizationPolicy policy = AuthUtility.CreateFunctionPolicy();
+            var httpTrigger = descriptor.HttpTriggerAttribute;
+            bool shouldAuthorize = !descriptor.Metadata.IsProxy && httpTrigger?.AuthLevel != AuthorizationLevel.Anonymous;
 
+            if (shouldAuthorize)
```

Comment:
Skip all of the auth level components if auth level is anonymous

Reply:
BTW - I see now that I should invert this logic so that rather than whitelisting, I determine when I can skip. That's safer. I.e. skip only when function is a proxy, or function is Anonymous w/o any of the special headers. I'll make that change.